### PR TITLE
Merge changes made for hass back into ToonHA

### PR DESCRIPTION
**Pull request by boltgolt**
As [this](https://github.com/krocat/ToonHA/tree/6c382986974a9adde0d6ea53dd607ece4ae8027a) version of ToonHA has finally been added into the main hass repo it's time to merge the changes requested by the hass team back into ToonHA. I've modified the changes so it fits into ToonHA as a custom component. (looks like i can't push to new branches so this is a PR instead of a MR)

@costastf could you elaborate what you meant with removing the fibaro part?

### CODE REVIEW

File: climate/toon.py [hunk 1]
```diff
@@ -40,8 +35,10 @@ def __init__(self, hass):
         self._state = None
         self._temperature = None
         self._setpoint = None
-        self._operation_list = [STATE_HEAT, STATE_HOME, STATE_NOT_HOME,
-                                STATE_COOL, STATE_UNKNOWN]
+        self._operation_list = [STATE_PERFORMANCE,
+                                STATE_HEAT,
+                                STATE_ECO,
+                                STATE_COOL]
```
Comment (boltgolt) [inline climate/toon.py:41]: I'm not very happy with this change, but it needed to be done to be able to merge. As the hass team pointed out, all new climate components need to use these states and they enable better control of toon by scripts/automations. Half way though the pull request some [new states](https://github.com/home-assistant/home-assistant/commit/312de6b3a369bdc58dda94b2a78167ef38821cee#diff-b9607e70a7ce29fdc462c9457fd38bbfR49) were added to the core, so i've used them to get as close as possible to the actual toon states. 

I can imagine you'd want to keep the normal native toon states in this repo though.

Comment (costastf) [reply climate/toon.py:41]: No, I don't mind at all. This is absolutely fine and it makes sense to use common states across common devices. The previous time we tried to get this in there, there were not enough states that made sense to describe the usability so basically toon was getting crippled.

**Comment by costastf** [comment]
#119 class FibaroSensor(Entity):  should probably be SmartPlug and 
#223 class FibaroSmokeDetector(Entity): should probably be SmokeDetector 
Any compatible device can be handled not only fibaro ones.

**Comment by costastf** [comment]
Lovely, thanks!

**Pull request merged**